application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -285,9 +282,14 @@ private void checkBusyStatusAllChannels(@NotNull JDA jda) {
      *
      * @param event the generic event that includes the 'onGuildMessageReceived'.
      */
+    @SuppressWarnings("squid:S2583") // False-positive about the if-else-instanceof, sonar thinks
+                                     // the second case is unreachable; but it passes without
+                                     // pattern-matching. Probably a bug in SonarLint with Java 17.
     @Override
     public void onEvent(@NotNull GenericEvent event) {
-        if (event instanceof GuildMessageReceivedEvent guildEvent) {
+        if (event instanceof ReadyEvent readyEvent) {
+            onReady(readyEvent);
+        } else if (event instanceof GuildMessageReceivedEvent guildEvent) {
             if (guildEvent.isWebhookMessage() || guildEvent.getAuthor().isBot()) {
                 return;
```

Comment:
I think we should put the whole `GuildMessageRevievedEvent` handling in a separate method. Why don't we just implement `MessageReciever` and have that block in the `onGuildMessageRecieved` method

Reply:
I agree, but this should not be done in this PR but in a dedicated PR dealing with the free-command. borgrel is already awaitng this to get merged to do exactly those changes.

For this PR, i only did like the bare minimum to integrate the existing code so that it still works as before. Any further change would likely be a bigger refactoring and also needs more testing.

Reply:
I realized that from our discussion in the discord.